pkgs/development/python-modules/rich/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, CommonMark
+, colorama
+, dataclasses
+, ipywidgets
+, poetry
+, pygments
+, typing-extensions
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "rich";
+  version = "8.0.0";
+
+  # tests not included in pypi tarball
+  src = fetchFromGitHub {
+    owner = "willmcgugan";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0hv27b22x7dbx1i7nzsd8y8fymmvdak2hcx9242jwk4c1a7jr151";
+  };
+  format = "pyproject";
+
+  buildInputs = [ poetry ];
```

generally poetry is just used for packaging
```suggestion
  nativeBuildInputs = [ poetry ];
```